### PR TITLE
Enable ORTModule with bypass input

### PR DESCRIPTION
**Pull request by SherlockNoMad**
def forward(self, input1, **bypass_input**):
            out1 = self.fc1_2(self.relu1(self.fc1_1(input1)))
            return out1, **bypass_input**

Such module is now supported together with torch nightly build.

### CODE REVIEW

**Review by mrry — APPROVED** [review]

**Comment by thiagocrepaldi** [comment]
Are we officially taking dependency on torch nightly or this is just getting ready for future torch releases?

**Review by thiagocrepaldi — COMMENTED** [review]
Are we officially taking dependency on torch nightly or this is just getting ready for future torch releases?

File: orttraining/orttraining/core/framework/gradient_graph_builder.cc [hunk 1]
```diff
@@ -56,17 +56,31 @@ GradientGraphBuilder::GradientGraphBuilder(Graph* graph,
...
+      bool is_graph_input = false;
+      for (const auto input : graph_inputs) {
+        if (input->Name() == name) {
+          is_graph_input = true;
+          break;
+        }
+      }
+
+      if (is_graph_input) {
+        LOGS(logger_, INFO) << "NodeArg " << name << " cannot find a producer node, "
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/core/framework/gradient_graph_builder.cc:79]: Couldn't we add the identity node right here? This code is already iterating over the all the outputs and the corner case is properly detected

To do that in the frontend we would have to iterate over onnx model after it is exported - adding identity in the pytorch model doesnt work as it is optimized out during export. It is probably not that expensive, but as this is already being done here anyway, it is cheaper and handier

**Comment by thiagocrepaldi** [comment]
@SherlockNoMad we can close this as #7694 was merged and includes this change

**Pull request closed without merging**